The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/9441677925/job/26002762775) [(plain)](https://github.com/rust-lang/rust/commit/6c0aea5bc64b5121631bb73243b432a203b37848/checks/26002762775/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Getting action download info
Download action repository 'msys2/setup-msys2@v2.22.0' (SHA:cc11e9188b693c2b100158c3322424c4cc1dadea)
Download action repository 'actions/checkout@v4' (SHA:a5ac7e51b41094c92402da3b24376905380afc29)
Download action repository 'actions/upload-artifact@v4' (SHA:65462800fd760344b1a7b4382951275a0abb4808)
Complete job name: PR - mingw-check-tidy
git config --global core.autocrlf false
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
---
COPY scripts/sccache.sh /scripts/
RUN sh /scripts/sccache.sh

COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt \
    && pip3 install virtualenv
COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---

#12 [5/8] COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
#12 DONE 0.0s

#13 [6/8] RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt     && pip3 install virtualenv
#13 0.430   Downloading binaryornot-0.4.4-py2.py3-none-any.whl (9.0 kB)
#13 0.443 Collecting boolean-py==4.0
#13 0.449   Downloading boolean.py-4.0-py3-none-any.whl (25 kB)
#13 0.458 Collecting chardet==5.1.0
---
#13 3.626 Building wheels for collected packages: reuse
#13 3.627   Building wheel for reuse (pyproject.toml): started
#13 3.958   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 3.959   Created wheel for reuse: filename=reuse-1.1.0-cp310-cp310-manylinux_2_35_x86_64.whl size=181117 sha256=f5f58750481f69515c2c0d1d503daf565e2565c370d07fc6aeb95fe3498b4269
#13 3.959   Stored in directory: /tmp/pip-ephem-wheel-cache-eqg7ombk/wheels/c2/3c/b9/1120c2ab4bd82694f7e6f0537dc5b9a085c13e2c69a8d0c76d
#13 3.962 Installing collected packages: boolean-py, binaryornot, setuptools, reuse, python-debian, markupsafe, license-expression, jinja2, chardet
#13 3.984   Attempting uninstall: setuptools
#13 3.985     Found existing installation: setuptools 59.6.0
#13 3.986     Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
---
#13 5.209   Downloading virtualenv-20.26.2-py3-none-any.whl (3.9 MB)
#13 5.271      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.9/3.9 MB 66.3 MB/s eta 0:00:00
#13 5.321 Collecting filelock<4,>=3.12.2
#13 5.325   Downloading filelock-3.14.0-py3-none-any.whl (12 kB)
#13 5.342 Collecting distlib<1,>=0.3.7
#13 5.349   Downloading distlib-0.3.8-py2.py3-none-any.whl (468 kB)
#13 5.357      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 468.9/468.9 KB 88.2 MB/s eta 0:00:00
#13 5.385 Collecting platformdirs<5,>=3.9.1
#13 5.390   Downloading platformdirs-4.2.2-py3-none-any.whl (18 kB)
#13 5.476 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 5.640 Successfully installed distlib-0.3.8 filelock-3.14.0 platformdirs-4.2.2 virtualenv-20.26.2
#13 DONE 5.7s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      217024 kB
DirectMap2M:     7122944 kB
DirectMap1G:    11534336 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
downloading https://ci-artifacts.rust-lang.org/rustc-builds-alt/6d94a8727522e21c4309deb7034f5093531a6a76/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/llvm-6d94a8727522e21c4309deb7034f5093531a6a76-true/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm
---
   Compiling tidy v0.1.0 (/checkout/src/tools/tidy)
    Finished `release` profile [optimized] target(s) in 26.99s
##[endgroup]
fmt check
##[error]Diff in /checkout/tests/run-make/llvm-outputs/rmake.rs at line 1:
 // test that directories get created when emitting llvm bitcode and IR
 
-use run_make_support::{rustc, run_in_tmpdir, cwd};
+use run_make_support::{cwd, run_in_tmpdir, rustc};
 
 fn main() {
 fn main() {
##[error]Diff in /checkout/tests/run-make/llvm-outputs/rmake.rs at line 7:
- let mut path_bc = PathBuf::new();
- let mut path_ir = PathBuf::new();
- run_in_tmpdir(|| {
-  let p = cwd();
-  path_bc = p.join("nonexistant_dir_bc");
-  path_ir = p.join("nonexistant_dir_ir");
-  rustc().input("-").stdin("fn main() {}")
-   .out_dir(&path_bc).arg("--emit=llvm-bc").run();
-  rustc().input("-").stdin("fn main() {}")
-   .out_dir(&path_ir).arg("--emit=llvm-ir").run();
-  assert!(path_bc.exists());
-  assert!(path_ir.exists());
- });
- assert!(!path_bc.exists());
- assert!(!path_ir.exists());
+    let mut path_bc = PathBuf::new();
+    let mut path_ir = PathBuf::new();
+    run_in_tmpdir(|| {
+        let p = cwd();
+        path_bc = p.join("nonexistant_dir_bc");
+        path_ir = p.join("nonexistant_dir_ir");
+        rustc().input("-").stdin("fn main() {}").out_dir(&path_bc).arg("--emit=llvm-bc").run();
+        rustc().input("-").stdin("fn main() {}").out_dir(&path_ir).arg("--emit=llvm-ir").run();
+        assert!(path_bc.exists());
+        assert!(path_ir.exists());
+    });
+    assert!(!path_bc.exists());
+    assert!(!path_ir.exists());
 
 
fmt error: Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/tests/codegen/static-relocation-model-msvc.rs" "/checkout/tests/codegen/mainsubprogramstart.rs" "/checkout/tests/codegen/vec-shrink-panik.rs" "/checkout/tests/codegen/debug-vtable.rs" "/checkout/tests/codegen/cold-call-declare-and-call.rs" "/checkout/tests/codegen/aarch64-struct-align-128.rs" "/checkout/tests/codegen/mir-inlined-line-numbers.rs" "/checkout/tests/codegen/step_by-overflow-checks.rs" "/checkout/tests/codegen/noalias-box.rs" "/checkout/tests/codegen/coercions.rs" "/checkout/tests/codegen/skip-mono-inside-if-false.rs" "/checkout/tests/codegen/dst-offset.rs" "/checkout/tests/codegen/unwind-and-panic-abort.rs" "/checkout/tests/codegen/internalize-closures.rs" "/checkout/tests/codegen/debug-column-msvc.rs" "/checkout/tests/run-make/symlinked-libraries/bar.rs" "/checkout/tests/run-make/symlinked-libraries/foo.rs" "/checkout/tests/run-make/pgo-gen-no-imp-symbols/test.rs" "/checkout/tests/run-make/archive-duplicate-names/bar.rs" "/checkout/tests/run-make/archive-duplicate-names/foo.rs" "/checkout/tests/codegen/sanitizer/no-sanitize.rs" "/checkout/tests/codegen/sanitizer/memtag-attr-check.rs" "/checkout/tests/run-make/core-no-oom-handling/rmake.rs" "/checkout/tests/run-make/c-static-rlib/bar.rs" "/checkout/tests/run-make/c-static-rlib/foo.rs" "/checkout/tests/codegen/sanitizer/kcfi/emit-kcfi-operand-bundle-itanium-cxx-abi-normalized.rs" "/checkout/tests/codegen/sanitizer/kcfi/emit-kcfi-operand-bundle-itanium-cxx-abi.rs" "/checkout/tests/codegen/sanitizer/kcfi/emit-kcfi-operand-bundle.rs" "/checkout/tests/codegen/sanitizer/kcfi/emit-type-metadata-trait-objects.rs" "/checkout/tests/codegen/sanitizer/kcfi/emit-kcfi-operand-bundle-itanium-cxx-abi-generalized.rs" "/checkout/tests/codegen/sanitizer/kcfi/emit-kcfi-operand-bundle-itanium-cxx-abi-normalized-generalized.rs" "/checkout/tests/codegen/sanitizer/kcfi/emit-kcfi-operand-bundle-attr-no-sanitize.rs" "/checkout/tests/codegen/sanitizer/kcfi/add-kcfi-flag.rs" "/checkout/tests/codegen/sanitizer/scs-attr-check.rs" "/checkout/tests/codegen/sanitizer/memory-track-origins.rs" "/checkout/tests/codegen/sanitizer/sanitizer-recover.rs" "/checkout/tests/codegen/sanitizer/safestack-attr-check.rs" "/checkout/tests/codegen/sanitizer/kasan-emits-instrumentation.rs" "/checkout/tests/run-make/rustdoc-scrape-examples-remap/examples/ex.rs" "/checkout/tests/run-make/rustdoc-scrape-examples-remap/rmake.rs" "/checkout/tests/run-make/rustdoc-scrape-examples-remap/scrape.rs" "/checkout/tests/run-make/rustdoc-scrape-examples-remap/src/lib.rs" "/checkout/tests/run-make/rustdoc-scrape-examples-remap/src/a.rs" "/checkout/tests/run-make/issue-28595/b.rs" "/checkout/tests/run-make/issue-28595/a.rs" "/checkout/tests/run-make/llvm-outputs/rmake.rs" "/checkout/tests/run-make/no-alloc-shim/foo.rs" "/checkout/tests/run-make/alloc-no-oom-handling/rmake.rs" "/checkout/tests/codegen/sanitizer/cfi/emit-type-metadata-id-itanium-cxx-abi-sequence-types.rs" "/checkout/tests/codegen/sanitizer/cfi/emit-type-checks.rs" "/checkout/tests/codegen/sanitizer/cfi/emit-type-metadata-id-itanium-cxx-abi-pointer-types.rs" "/checkout/tests/codegen/sanitizer/cfi/emit-type-metadata-id-itanium-cxx-abi-method-secondary-typeid.rs" "/checkout/tests/codegen/sanitizer/cfi/emit-type-metadata-itanium-cxx-abi-generalized.rs" "/checkout/tests/codegen/sanitizer/cfi/emit-type-metadata-attr-cfi-encoding.rs" "/checkout/tests/codegen/sanitizer/cfi/emit-type-checks-attr-no-sanitize.rs" "/checkout/tests/codegen/sanitizer/cfi/emit-type-metadata-id-itanium-cxx-abi-trait-types.rs" "/checkout/tests/codegen/sanitizer/cfi/emit-type-metadata-id-itanium-cxx-abi-drop-in-place.rs" "/checkout/tests/codegen/sanitizer/cfi/emit-type-metadata-id-itanium-cxx-abi-function-types.rs" "/checkout/tests/codegen/sanitizer/cfi/generalize-pointers.rs" "/checkout/tests/codegen/sanitizer/cfi/emit-type-metadata-trait-objects.rs" "/checkout/tests/codegen/sanitizer/cfi/add-canonical-jump-tables-flag.rs" "/checkout/tests/codegen/sanitizer/cfi/emit-type-metadata-id-itanium-cxx-abi-lifetimes.rs" "/checkout/tests/codegen/sanitizer/cfi/emit-type-metadata-id-itanium-cxx-abi-user-defined-types.rs" "/checkout/tests/codegen/global_asm.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Mon Jun 10 02:45:43 UTC 2024
  network time: Mon, 10 Jun 2024 02:45:43 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/src/tools/run-make-support/src/rustc.rs
+++ b/src/tools/run-make-support/src/rustc.rs
@@ -107,6 +107,13 @@ impl Rustc {
         self
     }
 
+    /// Specify path to the output directory. Equivalent to `--out-dir`` in rustc.
+    pub fn out_dir<P: AsRef<Path>>(&mut self, path: P) -> &mut Self {
+        self.cmd.arg("--out-dir");
+        self.cmd.arg(path.as_ref());
+        self
+    }
+
     /// This flag defers LTO optimizations to the linker.
     pub fn linker_plugin_lto(&mut self, option: &str) -> &mut Self {
         self.cmd.arg(format!("-Clinker-plugin-lto={option}"));

--- a/src/tools/tidy/src/allowed_run_make_makefiles.txt
+++ b/src/tools/tidy/src/allowed_run_make_makefiles.txt
@@ -122,7 +122,6 @@ run-make/link-framework/Makefile
 run-make/link-path-order/Makefile
 run-make/linkage-attr-on-static/Makefile
 run-make/llvm-ident/Makefile
-run-make/llvm-outputs/Makefile
 run-make/long-linker-command-lines-cmd-exe/Makefile
 run-make/long-linker-command-lines/Makefile
 run-make/longjmp-across-rust/Makefile

--- a/tests/run-make/llvm-outputs/Makefile
+++ b/tests/run-make/llvm-outputs/Makefile
@@ -1,5 +1,0 @@
-include ../tools.mk
-
-all:
-	echo 'fn main() {}' | $(BARE_RUSTC) - --out-dir=$(TMPDIR)/random_directory_that_does_not_exist_ir/ --emit=llvm-ir
-	echo 'fn main() {}' | $(BARE_RUSTC) - --out-dir=$(TMPDIR)/random_directory_that_does_not_exist_bc/ --emit=llvm-bc

--- a/tests/run-make/llvm-outputs/rmake.rs
+++ b/tests/run-make/llvm-outputs/rmake.rs
@@ -1,0 +1,20 @@
+// test that directories get created when emitting llvm bitcode and IR
+
+use run_make_support::{cwd, run_in_tmpdir, rustc};
+use std::path::PathBuf;
+
+fn main() {
+    let mut path_bc = PathBuf::new();
+    let mut path_ir = PathBuf::new();
+    run_in_tmpdir(|| {
+        let p = cwd();
+        path_bc = p.join("nonexistant_dir_bc");
+        path_ir = p.join("nonexistant_dir_ir");
+        rustc().input("-").stdin("fn main() {}").out_dir(&path_bc).emit("llvm-bc").run();
+        rustc().input("-").stdin("fn main() {}").out_dir(&path_ir).emit("llvm-ir").run();
+        assert!(path_bc.exists());
+        assert!(path_ir.exists());
+    });
+    assert!(!path_bc.exists());
+    assert!(!path_ir.exists());
+}

